### PR TITLE
feat(dsl): Add if strategy

### PR DESCRIPTION
**Pull request by nbrouand**

### CODE REVIEW

File: dsl/src/test/kotlin/com/chutneytesting/kotlin/junit/engine/ChutneyTest.kt [hunk 1]
```diff
@@ -149,4 +149,38 @@ class ChutneyTest {
...
+        return Scenario(title = "A scenario") {
+            Given("Two values") {
+                ContextPutTask(
+                    entries = mapOf(
+                        "value1" to "",
+                        "value2" to ""
+                    )
+                )
+            }
+            When("Updateing values or not depending of strategy") {
```
Comment (DelaunayAlex) [inline dsl/src/test/kotlin/com/chutneytesting/kotlin/junit/engine/ChutneyTest.kt:164]: Updating

Comment (DelaunayAlex) [inline dsl/src/test/kotlin/com/chutneytesting/kotlin/junit/engine/ChutneyTest.kt:164]: Updating

**Review by DelaunayAlex — CHANGES_REQUESTED** [review]

**Review by boddissattva — DISMISSED** [review]

File: dsl/src/main/kotlin/com/chutneytesting/kotlin/dsl/ChutneyScenarioDsl.kt [hunk 2]
```diff
@@ -82,6 +82,12 @@ open class ForStrategy(dataset: String = "dataset".spEL, index: String = "i") :
         const val TYPE: String = "for"
     }
 }
+open class IfStrategy(condition: String) :
+    Strategy(type = TYPE, parameters = mapOf("condition" to condition)) {
```
Comment (KarimGl) [inline dsl/src/main/kotlin/com/chutneytesting/kotlin/dsl/ChutneyScenarioDsl.kt:86]: May be directly type = "if"

File: dsl/src/test/kotlin/com/chutneytesting/kotlin/junit/engine/ChutneyTestEngineTest.kt [hunk 3]
```diff
@@ -116,7 +116,7 @@ private class ChutneyTestEngineTest {
         testEngine.discover(discoveryRequest, UniqueId.forEngine(testEngine.id))
 
         // 9 times for 9 scenarios in 8 tests methods in ChutneyTest and SameMethodNameClassTest
```
Comment (KarimGl) [inline dsl/src/test/kotlin/com/chutneytesting/kotlin/junit/engine/ChutneyTestEngineTest.kt:118]: update or remove comment

**Review by boddissattva — APPROVED** [review]

**Pull request merged**